### ISSUE TITLE
Support new ad placement locations on home page (part 2)

### ISSUE THREAD

**Issue by jaw6** [forem/forem#19258]
## User Story (Who / What / Why)
This is a spin-off from https://github.com/forem/forem/issues/19225 — both can (theoretically) work in parallel.

We want to update `DisplayAd` to support new `placement_areas` and update server-rendered home-page views to include these new areas for signed-out users. (Signed-in home-page views are rendered client-side and will update later.)

New `placement_areas`:
* [ ] `home_hero` - A large hero banner appearing in the `@hero_area` "Campaign banner" slot
* [ ] these options should be available at `/admin/customization/display_ads/new`
* [ ] these options should appear in API documentation for `display_ads`

Considerations:
* `hero` billboards should take up the full width of the content container, styled for larger text sizes.
* `hero` billboards _cannot_ target `tags`
* `hero` billboards are restricted to the `in_house` type
* The initial implementation is focused on the home page articles feed, but later iterations may add these billboards to tagged articles feeds as well.
* These types of billboards are not able to target `logged_in` users _for now_, but we don't need to codify this as a validation.
* `hero` billboards will likely cover some of the functionality previously provided by "Campaign HTML Variants", but in our initial approach, both models will continue for some time. "Campaign HTML Variants" have a "close out" feature, where a user can stop seeing _all_ variations with one click, this may be brought to `hero` billboards in the future, but is not part of the initial scope of work.
* do **not** remove Campaign HtmlVariants (some communities use it)
* :link: There's additional background discussion of these new placement areas in 📝 [this document](https://docs.google.com/document/d/1biJjAjLsz2EHSLntJKViKQjDIjb36SRdmdarmE9dAPU/edit#) and design work [in Figma](https://www.figma.com/file/9UfSjNYJs745w3Q8JCUqsL/Small-Wins---%E2%80%A8Core-Experience-%E2%9A%A1%EF%B8%8F?node-id=1623-7063&t=Qj7oeBVeo7LrkobH-4) and [Loom](https://www.loom.com/share/8e9a5cb821524e4890c8f6df43d8d1e9)

**Comment by jaw6** [comment]
<img width="1785" alt="Screen Shot 2023-03-23 at 17 23 36" src="https://user-images.githubusercontent.com/2077/227271288-8992dc08-f77b-4cfd-bb1b-1bba4530c555.png">

@pkfrank @anujbhavsar96 talking about the proposed designs today with @mirie and we realized that our drive for [consistent rendering across "building blocks"](https://github.com/forem/forem/milestone/9) pushes responsibility for these designs to the *ad*, not the rendering system, possibly making some of these designs difficult, maybe even impossible under a "building blocks" approach. That is to say, we might be able to achieve these designs by writing HTML/CSS in the ad itself, but we don't have the tools to support different built-in layout options in the ad system directly.

_Eventually_, it could be that we can support more sophisticated multi-column hero ad layouts via `liquid` (I believe @benhalpern had proposed this at some point), but it's not currently on the short-term road-map for [rendering consistency](https://github.com/forem/forem/milestone/9).

I'm going to put this issue as `blocked` while we confirm the design "building blocks" constraints, so that we don't start work on it before we're confident we can deliver a workable layout.

**Comment by anujbhavsar96** [comment]
@jaw6 just curious, if we replicate sidebar billboard and just stretch it edge to edge will that work?

I’m trying to understand our tech limitations. Is it that we can’t have customised text container for billboards?

**Comment by jaw6** [comment]
> I’m trying to understand our tech limitations. Is it that we can’t have customised text container for billboards?

@anujbhavsar96 Custom container isn't a problem, but the ad inside the container will need to be responsible for internal layout. 

This is a bit of a moving target — we are [actively working on changes in this area](https://github.com/forem/forem/milestone/9) — but the idea is that a Page, Article, or Billboard should render text in a "consistent way", rather than having specific rendering/layouts for each type of element. That said, there's a _lot_ of flexibility in what we allow in HTML tags and attributes, it's just that doing "fancy" designs this way requires the ad's author to write more HTML than "simple" Markdown. 

I _think_ these designs _might_ all be possible (with the right HTML), but the need to write more HTML into the ad itself was a limitation @mirie and I thought was worth calling out.

**Comment by jaw6** [comment]
@anujbhavsar96 
> Is there a way we can make it easy from the design end?

Simpler designs/layouts should be easier for ad-writers to implement? But maybe we could provide "templates" somehow? (HTML we _know_ works, just fill in the "body"?)

**Comment by anujbhavsar96** [comment]
By simpler design/layout, do you mean something like the one below? Similar to sidebar units? @jaw6 

<img width="835" alt="Screenshot 2023-03-27 at 5 36 46 PM" src="https://user-images.githubusercontent.com/84626054/227918671-0a8dc68e-988c-4e8f-9a31-360a619f19c7.png">

**Comment by jaw6** [comment]
@anujbhavsar96 Yes. I mean, that's the kind of thing we can do quite easily.

**Comment by anujbhavsar96** [comment]
We can probably hack around a bit. We can achieve styling like the one I proposed earlier for the hero billboard while keeping the implementation simple, as approved by Joshua.

**Things we will have to consider while creating billboards.**
1. Keep the text center aligned
2. Use break line tag so we don't end up using edge-to-edge text containers. It's not good practice to have lots of words in one line. It makes paragraphs difficult to read.

Since hero placement would be used by us only, we can keep this hack in mind while creating billboards until we have advanced markdowns.


Example markdown
<img width="483" alt="image" src="https://user-images.githubusercontent.com/84626054/228429157-a6a5f48e-cdab-4a16-a0fb-435ac2f535d3.png">

And use the HTML variant for any advanced designs. What are your thoughts on this @benhalpern / @pkfrank ?

**Comment by benhalpern** [comment]
@anujbhavsar96 that seems fine to me as an approach for now, understanding that we will want to follow up this project with a more long-term solutions on how to handle bespoke design scenarios in a scalable way.

I'd say the one thing we should focus on for getting this one right are font sizing and spacing with different screen sizes, and getting that right.

That is to say: Full width should have appropriately _large_ font size, but mobile width should likely be _only slightly larger than feed size_.

For example: At max width, `h1` should probably be `fs-5xl`, but at mobile width, probably just `fs-3xl`, with the cascade of other element sizes acting similarly. I think having a good answer for this early on will help us come out with the cleanest and most usable "v1" for a limited set of use-cases, and we can manage more advanced needs as its own project later.

This is the only location thus far which would lend itself to requiring break points in that way with regard to font size. Does that all make sense?

**Comment by anujbhavsar96** [comment]
@benhalpern Yeah totally makes sense. And it's good to take care of this early on.

We can probably follow whatever breakpoints we have decided for headers already.

E.g., We use `fs-5xl` for the **Post title** on the **Post page** on desktop, and it's `fs-3xl` for mobile, just as you explained above.

**Comment by mirie** [comment]
@benhalpern / @anujbhavsar96 / @jaw6: want to check on this to see if there are other open questions to work through. 

Is this issue focused on the single column / full width billboard? Also @jaw6 what are your thoughts on the typography treatment being discussed above?

**Comment by jaw6** [comment]
@mirie I can't find a recording, but in my notes, I was asking a question in refinement along the lines of "with the limited functionality we've constrained here, is this still worth pursuing?" and — again, would love to review the recording if available — but I noted @benhalpern answering that Product would evaluate before Eng moved onto building this?

> what are your thoughts on the typography treatment being discussed above?

It seems fine to me, in abstract, but I think seeing it will be more meaningful.

**Comment by mirie** [comment]
@benhalpern as the product assignee, can you check out @jaw6's last question? I want to know if we're going to prep to discuss at upcoming refinement on Monday. Thank you!

**Comment by benhalpern** [comment]
I removed the blocked label because we have an eyesight towards the other functionality relevant to making the most of this location.

More info here: https://docs.google.com/document/d/1A70cIVZ9yTfj-Kp20jW5m6D5Om5ltgREVBukAuKR8eg/edit#

**Comment by mirie** [comment]
Notes from refinement:
- If a hero Billboard exists and a campaign is live, do not show the Campaign. 
- Add a feature flag for our usage on Dev until we have a clearer roadmap for Campaign/HTML variants
- There is no way to close this Billboard (as part of this issue -- will be a future issue)
- No async rendering for this Billboard placement
